otelcolbuilder/cmd/configprovider.go

```diff
@@ -99,3 +113,40 @@ func makeMapProvidersMap(providers ...confmap.Provider) map[string]confmap.Provi
 	}
 	return ret
 }
+
+func validateConfigFlags(flags *flag.FlagSet) error {
+	// test configFlag and opAmpConfigFlag for mutual exclusion
+	config := flags.Lookup(configFlag)
+	if config == nil {
+		return nil
+	}
+	opAmp := flags.Lookup(opAmpConfigFlag)
+	if opAmp == nil {
+		return nil
+	}
+	configVal := config.Value.String()
+	opAmpVal := opAmp.Value.String()
+	if configVal != config.DefValue && opAmpVal != "" {
+		return fmt.Errorf("cannot use --%s and --%s flags together", configFlag, opAmpConfigFlag)
+	}
+	return nil
+}
```

Comment:
Wouldn't this check be easier to do on already parsed flag values? We extract the locations slice and the value of the opamp config in the calling function anyway, 
```suggestion
func validateConfigLocations(localConfigLocations []string, opampConfigLocation string) error {
	if len(locations) > 0 && opampConfigLocation != "" {
		return fmt.Errorf("cannot use --%s and --%s flags together", configFlag, opAmpConfigFlag)
	}
	return nil
}
```

Reply:
I'm afraid that --config has a default, so your suggestion doesn't quite work, as `len(locations) > 0` is always true.

Reply:
`--config` doesn't have a default: https://github.com/SumoLogic/sumologic-otel-collector/blob/5b028e2b8cb21a66652c5bbff649aa41b32ef00c/otelcolbuilder/cmd/flags.go#L52

Reply:
:thinking: You're right, I'm not sure why I thought that it did!